### PR TITLE
[CIS-1367] Remove unnecessary API calls from `ChannelVC` and `ChannelListVC `

### PR DESCRIPTION
**Pull request by tbarbugli**
### 🔗 Issue Link

CIS-1367

### 🎯 Goal

Fix sending multiple unnecessary pagination requests when `ChannelListVC` and `ChannelVC` are shown.
This PR makes sure that only 1 API call is made to load both `ChannelListVC` and `ChannelVC`

### 🎨 Changes

Before this PR `ChannelListVC` was calling the Query Channels endpoint 3 times!

**ChannelListVC**
- Pagination was incorrectly firing the request to load another page 
- The `make` method was calling the `setUp` method (which calls `synchronize`)

**ChannelVC**
- Pagination was incorrectly firing the request to load another page 

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift [hunk 1]
```diff
@@ -17,7 +17,7 @@ open class ChatChannelListVC: _ViewController,
     /// The `ChatChannelListController` instance that provides channels data.
     public var controller: ChatChannelListController!
 
-    private var loadingPreviousMessages: Bool = false
+    private var paginatingChannels: Bool = false
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:20]: nit: it's preffered to name boolean variables starting with `is`
```suggestion
    private var isLoadingMoreChannels: Bool = false
```

Comment (evsaev) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:20]: the same applies to channel vc: `loadingPreviousMessages -> isLoadingPreviousMessages`

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift [hunk 2]
```diff
@@ -85,8 +85,7 @@ open class ChatChannelListVC: _ViewController,
         
         // Set the Controller on the ViewController
         chatChannelListVC.controller = controller
-        chatChannelListVC.setUp()
```
Comment (evsaev) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:88]: This seems legit, the `setUp` get invoked automatically as part of the `ChatChannelListVC's` view lifecycle 🤔

@adamrushy can we make this change or there's a reason to have it?

**Review by evsaev — APPROVED** [review]
Looks good, just some small comments regarding naming 👍 Let's wait for the green light from Adam and we will be good to 🚢 Do we need a changelog entry for this, btw?

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift (hunk 2 above)
Comment (adamrushy) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:88]: Yeah, I need to check this... I specifically added because there was a bug with delegates not being assigned if I remember rightly 🤔

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift [hunk 3]
```diff
@@ -124,6 +123,10 @@ open class ChatChannelListVC: _ViewController,
             return
         }
 
+        guard collectionView.isTracking || collectionView.isDecelerating else {
```
Comment (polqf) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:126]: Seems like this same check is duplicated in ChatChannelVC, what about creating an extension over UICollectionView?

Comment (tbarbugli) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:126]: one VC uses tableview and the other collectionview

Comment (polqf) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:126]: Both are UIScrollView under the hood

**Comment by bielikb** [comment]
@tbarbugli please point this PR against the release candidate `release/4.6.0` + reflect this fix in the `Changelog`.

File: Examples/SlackClone/SlackMessageOptionsResolver.swift [hunk 4]
```diff
@@ -14,7 +14,9 @@ final class SlackMessageOptionsResolver: ChatMessageLayoutOptionsResolver {
         appearance: Appearance
     ) -> ChatMessageLayoutOptions {
         var options = super.optionsForMessage(at: indexPath, in: channel, with: messages, appearance: appearance)
-        options.remove([.flipped, .bubble, .timestamp, .avatar, .avatarSizePadding, .authorName, .threadInfo, .reactions])
+
+        [.flipped, .bubble, .timestamp, .avatar, .avatarSizePadding, .authorName, .threadInfo, .reactions]
+            .forEach { options.remove($0) }
```
Comment (nuno-vieira) [inline Examples/SlackClone/SlackMessageOptionsResolver.swift:19]: Why this change? Keep in mind, if you are getting a Warning, this was a mistake and it is fixed on 4.6.0 branch. Please revert this change

File: Examples/SlackClone/SlackMessageOptionsResolver.swift [hunk 5]
```diff
@@ -27,7 +29,7 @@ final class SlackMessageOptionsResolver: ChatMessageLayoutOptionsResolver {
         }()
 
         if isFirstInGroup {
-            options.insert([.avatar, .timestamp, .authorName])
+            [.avatar, .timestamp, .authorName].forEach { options.insert($0) }
```
Comment (nuno-vieira) [inline Examples/SlackClone/SlackMessageOptionsResolver.swift:32]: Same here

**Review by nuno-vieira — CHANGES_REQUESTED** [review]
Requesting change just to make sure we revert that code. Other than that LGTM 👍

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift (hunk 2 above)
Comment (adamrushy) [reply Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift:88]: Okay, I tested this and is all working as expected so we can proceed. I also checked by using the Storyboard implementation and it's all working as expected

`let chatList = DemoChannelListVC.make(with: controller, storyboard: UIStoryboard(name: "Main", bundle: .main), storyboardId: "ChannelList")`

**Review by adamrushy — COMMENTED** [review]
Tested across the `make` function and removing the setUp call, all is working as expected and I also checked the Storyboard implementation and is all good. Just a conflict and a few other questions to answer then good to go 👍🏼

File: Examples/SlackClone/SlackMessageOptionsResolver.swift (hunk 4 above)
Comment (tbarbugli) [reply Examples/SlackClone/SlackMessageOptionsResolver.swift:19]: I was getting a warning about that

**Review by nuno-vieira — APPROVED** [review]
LGTM! 🚀

File: Examples/SlackClone/SlackMessageOptionsResolver.swift (hunk 4 above)
Comment (evsaev) [reply Examples/SlackClone/SlackMessageOptionsResolver.swift]: ✅ Discarded this change

File: Examples/SlackClone/SlackMessageOptionsResolver.swift (hunk 5 above)
Comment (evsaev) [reply Examples/SlackClone/SlackMessageOptionsResolver.swift]: ✅ Discarded this change
